tests/drivers/uart/uart_async_api/src/test_uart_async_wide.c

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright (c) 2023 Jeroen van Dooren
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "test_uart.h"
+
+K_SEM_DEFINE(tx_wide_done, 0, 1);
+K_SEM_DEFINE(tx_wide_aborted, 0, 1);
+K_SEM_DEFINE(rx_wide_rdy, 0, 1);
+K_SEM_DEFINE(rx_wide_buf_released, 0, 1);
+K_SEM_DEFINE(rx_wide_disabled, 0, 1);
+
+static ZTEST_BMEM const struct device *const uart_dev =
+	DEVICE_DT_GET(UART_NODE);
+
+static void init_test(void)
+{
+	__ASSERT_NO_MSG(device_is_ready(uart_dev));
+	uart_rx_disable(uart_dev);
+	uart_tx_abort(uart_dev);
+	k_sem_reset(&tx_wide_done);
+	k_sem_reset(&tx_wide_aborted);
+	k_sem_reset(&rx_wide_rdy);
+	k_sem_reset(&rx_wide_buf_released);
+	k_sem_reset(&rx_wide_disabled);
+}
+
+#ifdef CONFIG_USERSPACE
```

Comment:
I don't see real added value testing `CONFIG_USERSPACE` here, I'd prefer you add `uart_async_chain_read_wide` to test `uart_rx_buf_rsp_u16`.
And maybe even remove `uart_async_single_read_wide` if `uart_async_chain_read_wide` can test all API s at once.